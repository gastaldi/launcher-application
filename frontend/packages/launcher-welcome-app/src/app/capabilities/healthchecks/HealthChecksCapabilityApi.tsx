import * as faker from 'faker';
import { HttpApi } from '../../../shared/utils/HttpApi';

export interface HealthChecksCapabilityApi {
  getLivenessAbsoluteUrl(): string;

  getReadinessAbsoluteUrl(): string;

  doGetLiveness(): Promise<{ content: string, time: number }>;

  doGetReadiness(): Promise<{ content: string, time: number }>;
}

export const HEALTHCHECKS_LIVENESS_PATH = '/health';
export const HEALTHCHECKS_READINESS_PATH = '/health';

class HttpHealthChecksCapabilityApi implements HealthChecksCapabilityApi {
  constructor(private readonly httpApi: HttpApi) {
  }

  public getLivenessAbsoluteUrl(): string {
    return this.httpApi.getAbsoluteUrl(HEALTHCHECKS_LIVENESS_PATH);
  }

  public async doGetLiveness(): Promise<{ content: string, time: number }> {
    const r = await this.httpApi.get<string>(HEALTHCHECKS_LIVENESS_PATH);
    return ({content: r, time: Date.now()});
  }

  public getReadinessAbsoluteUrl(): string {
    return this.httpApi.getAbsoluteUrl(HEALTHCHECKS_READINESS_PATH);
  }

  public async doGetReadiness(): Promise<{ content: string, time: number }> {
    const r = await this.httpApi.get<string>(HEALTHCHECKS_READINESS_PATH);
    return ({content: r, time: Date.now()});
  }
}

class MockHealthChecksCapabilityApi implements HealthChecksCapabilityApi {
  public getLivenessAbsoluteUrl(): string {
    return 'http://mocked.io/liveness';
  }

  public async doGetLiveness(): Promise<{ content: string, time: number }> {
    return ({content: faker.random.boolean() ? 'OK' : 'NOK', time: Date.now()});
  }

  public getReadinessAbsoluteUrl(): string {
    return 'http://mocked.io/readiness';
  }

  public async doGetReadiness(): Promise<{ content: string, time: number }> {
    return ({content: faker.random.boolean() ? 'OK' : 'NOK', time: Date.now()});
  }
}

export function newMockHealthChecksCapabilityApi(): HealthChecksCapabilityApi {
  return new MockHealthChecksCapabilityApi();
}

export const mockHealthChecksCapabilityApi: HealthChecksCapabilityApi = newMockHealthChecksCapabilityApi();

export function newHttpHealthChecksCapabilityApi(httpApi: HttpApi): HealthChecksCapabilityApi {
  return new HttpHealthChecksCapabilityApi(httpApi);
}
