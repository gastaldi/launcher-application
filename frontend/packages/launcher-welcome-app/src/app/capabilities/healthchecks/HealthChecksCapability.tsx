import { Grid, GridItem } from '@patternfly/react-core';
import * as React from 'react';
import { ExternalLink } from '../../../shared/components/ExternalLink';
import { useRequestsState, HttpRequest, RequestsConsole } from '../../../shared/components/HttpRequest';
import { RequestTitle } from '../../../shared/components/RequestTitle';
import CapabilityCard from '../../components/CapabilityCard';
import capabilitiesConfig from '../../config/capabilitiesConfig';
import { HEALTHCHECKS_LIVENESS_PATH, HEALTHCHECKS_READINESS_PATH, mockHealthChecksCapabilityApi } from './HealthChecksCapabilityApi';

export interface HealthChecksCapabilityProps {
}

export const HealthChecksApiContext = React.createContext(mockHealthChecksCapabilityApi);

export function HealthChecksCapability(props: HealthChecksCapabilityProps) {
  const api = React.useContext(HealthChecksApiContext);
  const [requests, addRequestEntry] = useRequestsState();
  const execGetLiveness = () => api.doGetLiveness();
  const execGetReadiness = () =>  api.doGetReadiness();
  return (
    <CapabilityCard module="healthchecks">
      <CapabilityCard.Title>{capabilitiesConfig.healthchecks.icon} {capabilitiesConfig.healthchecks.name}</CapabilityCard.Title>
      <CapabilityCard.Body>
        <Grid>
          <GridItem span={12}>
            <p>
              Health checks are HTTP endpoints that the codebase exposes for the purpose of monitoring.
              OpenShift will periodically hit these to determine that the container is functioning properly.
              If not, it may kill and/or restart the pod to maintain the targeted number of instances configured to be running.
              </p>
            <p>Your application has been set up with the following Health Checks.</p>
            <p>
              <ExternalLink
                href="https://docs.openshift.com/container-platform/latest/dev_guide/application_health.html"
              >
                Learn more
              </ExternalLink>
            </p>
          </GridItem>
          <CapabilityCard.Separator />
          <GridItem span={12} className="http-request-service">
            <RequestTitle>Liveness</RequestTitle>
          </GridItem>
          <HttpRequest
            name="Liveness check"
            method="GET"
            url={api.getLivenessAbsoluteUrl()}
            path={HEALTHCHECKS_LIVENESS_PATH}
            execute={execGetLiveness}
            onRequestResult={addRequestEntry}
          />
          <GridItem span={12} className="http-request-service">
            <RequestTitle>Readiness</RequestTitle>
          </GridItem>
          <HttpRequest
            name="Readiness check"
            method="GET"
            path={HEALTHCHECKS_READINESS_PATH}
            url={api.getReadinessAbsoluteUrl()}
            execute={execGetReadiness}
            onRequestResult={addRequestEntry}
          />
          <GridItem span={12}>
            <RequestsConsole name="HealthChecks" requests={requests} />
          </GridItem>
        </Grid>
      </CapabilityCard.Body>
    </CapabilityCard>
  );
}
