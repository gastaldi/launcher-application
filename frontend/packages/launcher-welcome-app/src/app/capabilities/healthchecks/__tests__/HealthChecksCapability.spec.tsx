import * as React from 'react';
import { fireEvent, render, act, cleanup } from 'react-testing-library';
import { newMockHealthChecksCapabilityApi } from '../HealthChecksCapabilityApi';
import { HealthChecksCapability, HealthChecksApiContext } from '../HealthChecksCapability';
import moment from 'moment-timezone';

jest.useFakeTimers();

moment.tz.setDefault('UTC');

afterEach(() => {
  console.log('cleanup()');
  cleanup();
});

describe('<HealthChecksCapability />', () => {
  it('check that initial render is correct', () => {
    const component = render(<HealthChecksCapability />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('check that readiness is working', async () => {
    const api = newMockHealthChecksCapabilityApi();
    const result = { content: 'OK', time: 1542793377 };
    const spy = jest.spyOn(api, 'doGetReadiness').mockResolvedValue(result);
    const Wrapper: React.FunctionComponent = (props) => (<HealthChecksApiContext.Provider value={api}>{props.children}</HealthChecksApiContext.Provider>)
    const component = render(<HealthChecksCapability />, { wrapper: Wrapper });
    fireEvent.click(component.getByLabelText('Execute Readiness check'));
    await act(async () => {
      await spy.mock.results[0];
    });
    expect(spy).toHaveBeenCalled();
    expect(component.getByLabelText(result.content));
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('check that liveness is working', async () => {
    const api = newMockHealthChecksCapabilityApi();
    const result = { content: 'OK', time: 1542793377 };
    const spy = jest.spyOn(api, 'doGetLiveness').mockResolvedValue(result);
    const Wrapper: React.FunctionComponent = (props) => (<HealthChecksApiContext.Provider value={api}>{props.children}</HealthChecksApiContext.Provider>)
    const component = render(<HealthChecksCapability />, { wrapper: Wrapper });
    fireEvent.click(component.getByLabelText('Execute Liveness check'));
    await act(async () => {
      await spy.mock.results[0];
    });
    expect(spy).toHaveBeenCalled();
    expect(component.getByLabelText(result.content));
    expect(component.asFragment()).toMatchSnapshot();
  });
});
