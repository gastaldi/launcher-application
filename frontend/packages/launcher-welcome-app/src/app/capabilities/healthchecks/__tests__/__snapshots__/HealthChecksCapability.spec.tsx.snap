// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HealthChecksCapability /> check that initial render is correct 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card capability-card"
    id="healthchecks-capability"
  >
    <div
      class="pf-c-card__header capability-card-header"
    >
      <svg
        aria-hidden="true"
        class="with-text"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M320.2 243.8l-49.7 99.4c-6 12.1-23.4 11.7-28.9-.6l-56.9-126.3-30 71.7H60.6l182.5 186.5c7.1 7.3 18.6 7.3 25.7 0L451.4 288H342.3l-22.1-44.2zM473.7 73.9l-2.4-2.5c-51.5-52.6-135.8-52.6-187.4 0L256 100l-27.9-28.5c-51.5-52.7-135.9-52.7-187.4 0l-2.4 2.4C-10.4 123.7-12.5 203 31 256h102.4l35.9-86.2c5.4-12.9 23.6-13.2 29.4-.4l58.2 129.3 49-97.9c5.9-11.8 22.7-11.8 28.6 0l27.6 55.2H481c43.5-53 41.4-132.3-7.3-182.1z"
          transform=""
        />
      </svg>
       Health Checks
    </div>
    <div
      class="pf-c-card__body capability-card-body"
    >
      <div
        class="pf-l-grid"
      >
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <p>
            Health checks are HTTP endpoints that the codebase exposes for the purpose of monitoring. OpenShift will periodically hit these to determine that the container is functioning properly. If not, it may kill and/or restart the pod to maintain the targeted number of instances configured to be running.
          </p>
          <p>
            Your application has been set up with the following Health Checks.
          </p>
          <p>
            <a
              href="https://docs.openshift.com/container-platform/latest/dev_guide/application_health.html"
              target="_blank"
            >
              Learn more 
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
                  transform=""
                />
              </svg>
            </a>
          </p>
        </div>
        <div
          class="capability-card-separator"
        >
          <hr />
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col http-request-service"
        >
          <svg
            aria-hidden="true"
            class="with-text"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              transform=""
            />
          </svg>
           Liveness
        </div>
        <div
          class="http-request method-get"
        >
          <div
            class="definition"
          >
            <div
              class="http-request-def"
            >
              <span
                class="http-request-method"
              >
                GET
              </span>
               
              <span
                class="http-request-path"
              >
                /health
              </span>
              <button
                class="pf-c-button pf-m-primary copy-button"
                title="Copy as curl"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 384 512"
                  width="1em"
                >
                  <path
                    d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              aria-label="Execute Liveness check"
              class="pf-c-button pf-m-primary http-request-button"
              title="Execute Liveness check"
              type="button"
            >
              Execute
            </button>
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col http-request-service"
        >
          <svg
            aria-hidden="true"
            class="with-text"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              transform=""
            />
          </svg>
           Readiness
        </div>
        <div
          class="http-request method-get"
        >
          <div
            class="definition"
          >
            <div
              class="http-request-def"
            >
              <span
                class="http-request-method"
              >
                GET
              </span>
               
              <span
                class="http-request-path"
              >
                /health
              </span>
              <button
                class="pf-c-button pf-m-primary copy-button"
                title="Copy as curl"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 384 512"
                  width="1em"
                >
                  <path
                    d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              aria-label="Execute Readiness check"
              class="pf-c-button pf-m-primary http-request-button"
              title="Execute Readiness check"
              type="button"
            >
              Execute
            </button>
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <div
            aria-label="HealthChecks Console"
            class="console"
          >
            <div
              class="console-content no-content"
            >
              <span
                class="placeholder"
              >
                <span
                  class="prefix"
                >
                  &gt;_
                </span>
                Â  Use the 'Execute' button(s) above to get things started in this console...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`<HealthChecksCapability /> check that liveness is working 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card capability-card"
    id="healthchecks-capability"
  >
    <div
      class="pf-c-card__header capability-card-header"
    >
      <svg
        aria-hidden="true"
        class="with-text"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M320.2 243.8l-49.7 99.4c-6 12.1-23.4 11.7-28.9-.6l-56.9-126.3-30 71.7H60.6l182.5 186.5c7.1 7.3 18.6 7.3 25.7 0L451.4 288H342.3l-22.1-44.2zM473.7 73.9l-2.4-2.5c-51.5-52.6-135.8-52.6-187.4 0L256 100l-27.9-28.5c-51.5-52.7-135.9-52.7-187.4 0l-2.4 2.4C-10.4 123.7-12.5 203 31 256h102.4l35.9-86.2c5.4-12.9 23.6-13.2 29.4-.4l58.2 129.3 49-97.9c5.9-11.8 22.7-11.8 28.6 0l27.6 55.2H481c43.5-53 41.4-132.3-7.3-182.1z"
          transform=""
        />
      </svg>
       Health Checks
    </div>
    <div
      class="pf-c-card__body capability-card-body"
    >
      <div
        class="pf-l-grid"
      >
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <p>
            Health checks are HTTP endpoints that the codebase exposes for the purpose of monitoring. OpenShift will periodically hit these to determine that the container is functioning properly. If not, it may kill and/or restart the pod to maintain the targeted number of instances configured to be running.
          </p>
          <p>
            Your application has been set up with the following Health Checks.
          </p>
          <p>
            <a
              href="https://docs.openshift.com/container-platform/latest/dev_guide/application_health.html"
              target="_blank"
            >
              Learn more 
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
                  transform=""
                />
              </svg>
            </a>
          </p>
        </div>
        <div
          class="capability-card-separator"
        >
          <hr />
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col http-request-service"
        >
          <svg
            aria-hidden="true"
            class="with-text"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              transform=""
            />
          </svg>
           Liveness
        </div>
        <div
          class="http-request method-get"
        >
          <div
            class="definition"
          >
            <div
              class="http-request-def"
            >
              <span
                class="http-request-method"
              >
                GET
              </span>
               
              <span
                class="http-request-path"
              >
                /health
              </span>
              <button
                class="pf-c-button pf-m-primary copy-button"
                title="Copy as curl"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 384 512"
                  width="1em"
                >
                  <path
                    d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              aria-label="Execute Liveness check"
              class="pf-c-button pf-m-primary http-request-button"
              title="Execute Liveness check"
              type="button"
            >
              Execute
            </button>
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col http-request-service"
        >
          <svg
            aria-hidden="true"
            class="with-text"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              transform=""
            />
          </svg>
           Readiness
        </div>
        <div
          class="http-request method-get"
        >
          <div
            class="definition"
          >
            <div
              class="http-request-def"
            >
              <span
                class="http-request-method"
              >
                GET
              </span>
               
              <span
                class="http-request-path"
              >
                /health
              </span>
              <button
                class="pf-c-button pf-m-primary copy-button"
                title="Copy as curl"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 384 512"
                  width="1em"
                >
                  <path
                    d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              aria-label="Execute Readiness check"
              class="pf-c-button pf-m-primary http-request-button"
              title="Execute Readiness check"
              type="button"
            >
              Execute
            </button>
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <div
            aria-label="HealthChecks Console"
            class="console"
          >
            <div
              class="console-content expanded"
            >
              <span
                class="placeholder"
              >
                <span
                  class="prefix"
                >
                  &gt;_
                </span>
                Â  Use the 'Execute' button(s) above to get things started in this console...
              </span>
              <div
                class="console-content-row"
              >
                <div>
                  <span
                    class="prefix"
                  >
                    $
                  </span>
                  Â 
                  <span
                    class="time"
                  >
                    8:33:13 PM
                  </span>
                  Â 
                  <span
                    class="method method-get"
                  >
                    GET
                  </span>
                  Â 
                  <span
                    class="url"
                  >
                    http://mocked.io/liveness
                  </span>
                  :
                </div>
                <div
                  aria-label="OK"
                >
                  OK
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`<HealthChecksCapability /> check that readiness is working 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card capability-card"
    id="healthchecks-capability"
  >
    <div
      class="pf-c-card__header capability-card-header"
    >
      <svg
        aria-hidden="true"
        class="with-text"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M320.2 243.8l-49.7 99.4c-6 12.1-23.4 11.7-28.9-.6l-56.9-126.3-30 71.7H60.6l182.5 186.5c7.1 7.3 18.6 7.3 25.7 0L451.4 288H342.3l-22.1-44.2zM473.7 73.9l-2.4-2.5c-51.5-52.6-135.8-52.6-187.4 0L256 100l-27.9-28.5c-51.5-52.7-135.9-52.7-187.4 0l-2.4 2.4C-10.4 123.7-12.5 203 31 256h102.4l35.9-86.2c5.4-12.9 23.6-13.2 29.4-.4l58.2 129.3 49-97.9c5.9-11.8 22.7-11.8 28.6 0l27.6 55.2H481c43.5-53 41.4-132.3-7.3-182.1z"
          transform=""
        />
      </svg>
       Health Checks
    </div>
    <div
      class="pf-c-card__body capability-card-body"
    >
      <div
        class="pf-l-grid"
      >
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <p>
            Health checks are HTTP endpoints that the codebase exposes for the purpose of monitoring. OpenShift will periodically hit these to determine that the container is functioning properly. If not, it may kill and/or restart the pod to maintain the targeted number of instances configured to be running.
          </p>
          <p>
            Your application has been set up with the following Health Checks.
          </p>
          <p>
            <a
              href="https://docs.openshift.com/container-platform/latest/dev_guide/application_health.html"
              target="_blank"
            >
              Learn more 
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
                  transform=""
                />
              </svg>
            </a>
          </p>
        </div>
        <div
          class="capability-card-separator"
        >
          <hr />
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col http-request-service"
        >
          <svg
            aria-hidden="true"
            class="with-text"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              transform=""
            />
          </svg>
           Liveness
        </div>
        <div
          class="http-request method-get"
        >
          <div
            class="definition"
          >
            <div
              class="http-request-def"
            >
              <span
                class="http-request-method"
              >
                GET
              </span>
               
              <span
                class="http-request-path"
              >
                /health
              </span>
              <button
                class="pf-c-button pf-m-primary copy-button"
                title="Copy as curl"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 384 512"
                  width="1em"
                >
                  <path
                    d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              aria-label="Execute Liveness check"
              class="pf-c-button pf-m-primary http-request-button"
              title="Execute Liveness check"
              type="button"
            >
              Execute
            </button>
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col http-request-service"
        >
          <svg
            aria-hidden="true"
            class="with-text"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              transform=""
            />
          </svg>
           Readiness
        </div>
        <div
          class="http-request method-get"
        >
          <div
            class="definition"
          >
            <div
              class="http-request-def"
            >
              <span
                class="http-request-method"
              >
                GET
              </span>
               
              <span
                class="http-request-path"
              >
                /health
              </span>
              <button
                class="pf-c-button pf-m-primary copy-button"
                title="Copy as curl"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 384 512"
                  width="1em"
                >
                  <path
                    d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="action"
          >
            <button
              aria-label="Execute Readiness check"
              class="pf-c-button pf-m-primary http-request-button"
              title="Execute Readiness check"
              type="button"
            >
              Execute
            </button>
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <div
            aria-label="HealthChecks Console"
            class="console"
          >
            <div
              class="console-content expanded"
            >
              <span
                class="placeholder"
              >
                <span
                  class="prefix"
                >
                  &gt;_
                </span>
                Â  Use the 'Execute' button(s) above to get things started in this console...
              </span>
              <div
                class="console-content-row"
              >
                <div>
                  <span
                    class="prefix"
                  >
                    $
                  </span>
                  Â 
                  <span
                    class="time"
                  >
                    8:33:13 PM
                  </span>
                  Â 
                  <span
                    class="method method-get"
                  >
                    GET
                  </span>
                  Â 
                  <span
                    class="url"
                  >
                    http://mocked.io/readiness
                  </span>
                  :
                </div>
                <div
                  aria-label="OK"
                >
                  OK
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
